tests/std/tests/P0355R7_calendars_and_time_zones_clocks/test.cpp

```diff
@@ -0,0 +1,477 @@
+    static_assert(leap_second{sys_seconds{42s}, false, 0s}.value() == -1s);
+
+    return true;
+}
+
+constexpr bool operator==(const leap_second_info& lhs, const leap_second_info& rhs) {
+    return lhs.is_leap_second == rhs.is_leap_second && lhs.elapsed == rhs.elapsed;
+}
+
+template <class DurationRep>
+void test_leap_second_info(const leap_second& leap, seconds accum) {
+    const bool is_positive = (leap.value() == 1s);
+    // First UTC time when leap is counted, before insertion of a positive leap, after insertion of a negative one.
+    const utc_time<duration<DurationRep>> utc_leap{leap.date().time_since_epoch() + accum + (is_positive ? 0s : -1s)};
+
+    auto lsi = get_leap_second_info(utc_leap - 1s);
+    assert(lsi == (leap_second_info{false, accum}));
+
+    lsi = get_leap_second_info(utc_leap - 500ms);
+    assert(lsi == (leap_second_info{false, accum}));
+
+    accum += leap.value();
+    lsi = get_leap_second_info(utc_leap);
+    assert(lsi == (leap_second_info{is_positive, accum}));
+
+    lsi = get_leap_second_info(utc_leap + 500ms);
+    assert(lsi == (leap_second_info{is_positive, accum}));
+
+    lsi = get_leap_second_info(utc_leap + 1s);
+    assert(lsi == (leap_second_info{false, accum}));
+}
+
+template <class Duration>
+void test_utc_clock_to_sys(const leap_second& leap) {
+    sys_time<Duration> before_leap;
+    if constexpr (is_integral_v<typename Duration::rep>) {
+        before_leap = leap.date() - Duration{1};
+    } else {
+        before_leap =
+            sys_time<Duration>{Duration{nextafter(leap.date().time_since_epoch().count(), typename Duration::rep{0})}};
+    }
+
+    auto u = utc_clock::from_sys(before_leap); // just before leap second
+    assert(utc_clock::from_sys(utc_clock::to_sys(u)) == u);
+    if (leap.value() == 1s) {
+        u += Duration{1};
+        assert(utc_clock::to_sys(u) == before_leap); // during
+    } else {
+        assert(utc_clock::from_sys(utc_clock::to_sys(u)) == u);
+    }
+
+    u += 1s;
+    assert(utc_clock::from_sys(utc_clock::to_sys(u)) == u); // just after
+}
+
+template <class Duration>
+void test_file_clock_from_utc(const leap_second& leap) {
+    static_assert(is_same_v<filesystem::file_time_type::clock, file_clock>);
+
+    const auto file_leap = clock_cast<file_clock>(leap.date());
+
+    auto u = utc_clock::from_sys(leap.date() - Duration{1}); // just before leap second
+    assert(file_clock::to_utc(file_clock::from_utc(u)) == u);
+
+    if (leap.value() == 1s && leap.date() <= sys_days{1d / January / 2017y}) {
+        u += Duration{1};
+        assert(file_clock::from_utc(u) == file_leap - Duration{1}); // during
+    } else {
+        assert(file_clock::to_utc(file_clock::from_utc(u)) == u);
+    }
+
+    u += 1s;
+    assert(file_clock::to_utc(file_clock::from_utc(u)) == u); // just after
+}
+
+void test_utc_clock_from_sys(const leap_second& leap, seconds offset) {
+    // Generalized from [time.clock.utc.members]/3 Example 1.

```

Comment:
This should cite the WP number (as paragraphs are renumbered over time, and occasionally more significant edits happen). Occurs once below.